### PR TITLE
Use ResourceDropdown component for Buckets page 

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
Screenshot from 2022-07-04 16-32-42
Screenshot from 2022-07-04 16-32-54

### CODE REVIEW

File: packages/components/data-resource/s3-endpoint-type.tsx [hunk 1]
```diff
@@ -60,15 +60,13 @@ export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
   };
 
   const getSecrets = React.useCallback(
-    (e) =>
+    (resorce) =>
```
Comment (debjyoti-pandit) [inline packages/components/data-resource/s3-endpoint-type.tsx:63]: though it is being changed [here](https://github.com/red-hat-storage/mcg-ms-console/pull/39).
resorce -> resource

File: packages/components/bucket-policy/bucket-policy-body.tsx [hunk 2]
```diff
@@ -95,6 +99,23 @@ export const BucketPolicyBody: React.FC<BucketPolicyBodyProps> = ({
   const isSingleDataReource = state.dataResourceType === BucketClassType.SINGLE;
   const showDataResource = dataResourcesLoaded && !dataResourcesError;
 
+  const getInitialSelection = React.useCallback(
+    (resources: K8sResourceCommon[]) =>
+      resources.find(
+        (resource) => resource.metadata.name === DATA_FEDERATION_NAMESPACE
```
[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply packages/components/bucket-policy/bucket-policy-body.tsx:105]: we still be needing getInitialSelection function... both are same thing `DATA_FEDERATION_NAMESPACE` or `bucketPolicyInitialState.obcNamespace`

[the remark this answers is not in the document]
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/bucket-policy-body.tsx:105]: why? do we need to have obcNamespace set as an empty string somewhere?

[the remark this answers is not in the document]
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/bucket-policy-body.tsx:105]: yeah, that I got it. I didn't mean to remove the `getInitialSelection` function. I meant keeping the initial state in one place.

[the remark this answers is not in the document]
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/bucket-policy-body.tsx:105]: `
const getInitialSelection = React.useCallback(
    (resources: K8sResourceCommon[]) =>
      resources.find(
        (resource) => resource.metadata.name === bucketPolicyInitialState.obcNamespace
'

[the remark this answers is not in the document]
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/bucket-policy-body.tsx:105]: and in `bucketPolicyInitialState.obcNamespace` we will have `DATA_FEDERATION_NAMESPACE`

**Pull request merged**